pkg/istio/onboarding/controller_test.go

```diff
@@ -0,0 +1,401 @@
+	assert.Equal(t, configStoreCache, c.store, "config store cache pointer should be equal")
+}
+
+func TestProcessEvent(t *testing.T) {
+	c := &Controller{queue: workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter())}
+
+	tests := []struct {
+		name             string
+		inputObject      interface{}
+		expectedQueueLen int
+		expectedKeyName  string
+	}{
+		{
+			name:             "test processing a service object",
+			inputObject:      onboardedService,
+			expectedQueueLen: 1,
+			expectedKeyName:  "test-namespace/onboarded-service",
+		},
+		{
+			name:             "test processing a non service object",
+			inputObject:      nil,
+			expectedQueueLen: 0,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c.processEvent(cache.MetaNamespaceKeyFunc, onboardedService)
+			assert.Equal(t, tt.expectedQueueLen, c.queue.Len(), "expected queue length should be equal")
+
+			if tt.expectedQueueLen > 0 {
+				item, shutdown := c.queue.Get()
+				assert.Equal(t, tt.expectedKeyName, item, "expected key name should be equal")
+				assert.Equal(t, false, shutdown, "shutdown should be false")
+			}
+		})
+	}
+
+	c.queue.ShutDown()
+}
+
+func TestAddService(t *testing.T) {
+	c := &Controller{}
+
+	tests := []struct {
+		name              string
+		inputServices     []string
+		clusterRbacConfig *v1alpha1.RbacConfig
+		expectedArray     []string
+	}{
+		{
+			name:              "test adding new service to ClusterRbacConfig",
+			inputServices:     []string{onboardedServiceName},
+			clusterRbacConfig: createClusterRbacConfig(),
+			expectedArray:     []string{onboardedServiceName},
+		},
+		{
+			name:              "test adding existing service to ClusterRbacConfig",
+			inputServices:     []string{onboardedServiceName},
+			clusterRbacConfig: createClusterRbacConfig(existingServiceName),
+			expectedArray:     []string{existingServiceName, onboardedServiceName},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c.addServices(tt.inputServices, tt.clusterRbacConfig)
+			assert.Equal(t, tt.expectedArray, tt.clusterRbacConfig.Inclusion.Services, "ClusterRbacConfig service list should contain expected services")
+		})
+	}
+}
+
+func TestDeleteService(t *testing.T) {
+	c := &Controller{}
+
+	tests := []struct {
+		name              string
+		clusterRbacConfig *v1alpha1.RbacConfig
+		inputArray        []string
+		expectedArray     []string
+	}{
+		{
+			name:              "test deleting service from ClusterRbacConfig",
+			clusterRbacConfig: createClusterRbacConfig(onboardedServiceName),
+			inputArray:        []string{onboardedServiceName},
+			expectedArray:     []string{},
+		},
+		{
+			name:              "test deleting existing service from ClusterRbacConfig",
+			clusterRbacConfig: createClusterRbacConfig(onboardedServiceName, existingServiceName),
+			inputArray:        []string{onboardedServiceName},
+			expectedArray:     []string{existingServiceName},
```

Comment:
could we add test cases for empty inputArray, and a servce that doesn't exist on the clusterRbacConfig

Reply:
Updated.